galasa-ui/src/components/headers/ThemeSelector.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+'use client';
+
+import React, { useTransition } from "react";
+import styles from "@/styles/Selector.module.css"; 
+import { ThemeType, useTheme } from "@/contexts/ThemeContext";
+import { Sun, Moon, Laptop } from '@carbon/icons-react';
+import { Tooltip } from '@carbon/react';
+
+const themeOptions: { id: ThemeType; label: string; icon: React.ReactNode; tooltip: string }[] = [
+  { id: 'light', label: 'Light', icon: <Sun size={20} />, tooltip: 'Switch to light mode' },
+  { id: 'dark', label: 'Dark', icon: <Moon size={20} />, tooltip: 'Switch to dark mode' },
+  { id: 'system', label: 'System', icon: <Laptop size={20} />, tooltip: 'Switch to system preference' },
+];
+
+export default function ThemeSelector() {
+  const { theme, setTheme } = useTheme();
+  const [isPending, startTransition] = useTransition();
+  const idx = themeOptions.findIndex((o) => o.id === theme);
+  const currentTheme = themeOptions[idx] || themeOptions[0];
+  const next = themeOptions[(idx + 1) % themeOptions.length];
+
+  const cycleTheme = () => {
+    startTransition(() => {
+      setTheme(next.id as ThemeType);
+    });
+  };
+
+  return (
+    <div className={styles.themeSwitcher}>
+      <Tooltip label={next.tooltip} align="bottom">
```

Comment:
Thanks for adding this. However, the issue mentioned in https://github.com/galasa-dev/webui/pull/127#discussion_r2177956256 regarding the tooltip having a light theme regardless of the global theme remains.

Light mode (it's not obvious that this is a tooltip):
<img width="166" alt="Screenshot 2025-07-04 at 11 56 07" src="https://github.com/user-attachments/assets/fcfecc20-aaf6-4f0b-b479-6d97cb21e5ee" />

Dark mode (looks good! Tooltip is clearly visible):
<img width="260" alt="Screenshot 2025-07-04 at 11 59 33" src="https://github.com/user-attachments/assets/f5107492-e7da-427e-bec2-3eb4ef865ef6" />


In light mode, it would be good if the tooltip could appear more clearly by having a dark background and white text.